### PR TITLE
DD-1336 Finish-dd-manage-deposit

### PR DESCRIPTION
**Pull request by aliassheikh**
Fixes DD-1336 Finish-dd-manage-deposit

# Description of changes
- Added finalised column names to the `deposit_properties` table
- Changed the code basis on new columns (DepositProperties class)
- The absolute path as location of deposits couldn't resolve in the server side completely and/or correctly. The functions parameter `Path` are substituted with `File`
- The string limits defined for `location` and `description` columns + code changes for these
- `poolingInterval` is now configurable via service's configuration file + code changes for this 
- `deleted` column is protected now and delete notification avoided during moving of the deposit folders
- Adapted the rest of code according the changes

# How to test
- Virtaul box -vagrant - deploy (dd-manage-depost) :  `https://dar.dans.knaw.nl/dataverse/root/`
- add SWORD2 zipped bags:
**Ex.**. `curl -v -i -X POST -H "Content-Disposition: attachment; filename= test-bag-1.zip" -H 'Content-Type: application/zip' -H "Content-MD5: $(md5 -q test-bag-1.zip)" -H 'Packaging: http://purl.org/net/sword/package/BagIt' --data-binary "@test-bag-1.zip" -u user001:user001 http://dar.dans.knaw.nl:20320/collection/1`

- look at / follow the database `report` via:
  - on browser: `http://dar.dans.knaw.nl:20355/report`
  -  as json records: `curl -X GET http://dar.dans.knaw.nl:20355/report`
  - as csv text: `curl -X GET -H 'Accept: text/csv' http://dar.dans.knaw.nl:20355/report`
- look at the folders via vagrant ssh: `tree -L 3 /var/opt/dans.knaw.nl/tmp/auto-ingest/`
- change `deposits.properties` using `nano` ==> the records should be refreshed (specially `depositUpdateTimestamp`) 
- move the deposits using `mv` command ==> `location` column should show the last deposit path
- delete the deposits using `rm -fr` command ==> `deleted` column should changed from `false` into `true`
- delete records using filters, such as:
  - `curl -X POST http://dar.dans.knaw.nl:20355/delete-deposit\?deleted\=true`
  - `curl -X POST http://dar.dans.knaw.nl:20355/delete-deposit\?user=user001` 

# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/main/java/nl/knaw/dans/managedeposit/core/DepositProperties.java [hunk 1]
```diff
@@ -140,11 +163,11 @@ public boolean equals(Object o) {
...
         if (!depositId.equals(that.depositId))
             return false;
-        return createdDate.equals(that.createdDate);
+        return depositCreationTimestamp.equals(that.depositCreationTimestamp);
     }
 
     @Override
     public int hashCode() {
-        return 31 * depositId.hashCode() + createdDate.hashCode();
+        return 31 * depositId.hashCode() + depositCreationTimestamp.hashCode();
```
Comment (jo-pol) [inline src/main/java/nl/knaw/dans/managedeposit/core/DepositProperties.java]: `depositCreationTimestamp` has no `nullable = false` annotation. Please avoid a NullPointerException here.

File: src/main/java/nl/knaw/dans/managedeposit/core/service/DepositStatusUpdater.java [hunk 2]
```diff
@@ -35,23 +38,47 @@ public DepositStatusUpdater(DepositPropertiesDAO depositPropertiesDAO) {
...
     @UnitOfWork
-    public void onCreateDeposit(Path depositPropertiesPath) {
-        Optional<DepositProperties> dpObject = depositPropertiesAssembler.assembleObject(depositPropertiesPath, false);
-        dpObject.ifPresent(depositPropertiesDAO::save);
+    public void onCreateDeposit(File depositPropertiesFile) {
+        Optional<DepositProperties> dp = depositPropertiesDAO.findById(depositPropertiesFile.getParentFile().getName());
+
+        if (dp.isPresent()) {
+            // The 'move deposit' action is processed in two steps: 1. `create` deposit in the new location; 2. `delete` it from the old location. Update the location column.
+//            Path currentFolder = depositPropertiesPath.getName(depositPropertiesPath.getNameCount() - 3);
```
Comment (jo-pol) [inline src/main/java/nl/knaw/dans/managedeposit/core/service/DepositStatusUpdater.java]: Could there be multiple threads or processes trying to perform this move on the same deposit? Then you can run into race conditions.

File: src/main/java/nl/knaw/dans/managedeposit/core/service/Limiter.java [hunk 3]
```diff
@@ -13,8 +13,21 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package nl.knaw.dans.managedeposit.core;
+package nl.knaw.dans.managedeposit.core.service;
+
+public class Limiter {
```
Comment (jo-pol) [inline src/main/java/nl/knaw/dans/managedeposit/core/service/Limiter.java]: Not sure I'm happy with this name. The mathematical meaning of truncate would be limited to `stripEnd`.
But googeling for definitions I also find "_You could truncate an essay by omitting a paragraph or two_" so `stripBegin` (and even stripCenter) also fits in the meaning of truncate. A synonym of the verb truncate is shorten.

File: src/main/java/nl/knaw/dans/managedeposit/core/DepositProperties.java (hunk 1 above)
Comment (aliassheikh) [reply src/main/java/nl/knaw/dans/managedeposit/core/DepositProperties.java]: Good Point 👍🏻

File: src/main/java/nl/knaw/dans/managedeposit/core/service/Limiter.java (hunk 3 above)
Comment (aliassheikh) [reply src/main/java/nl/knaw/dans/managedeposit/core/service/Limiter.java]: Indeed, it is a good idea to help 'findability' of the classes and functions. 'limiter' changed to 'TextTruncation'

File: src/main/java/nl/knaw/dans/managedeposit/core/service/DepositStatusUpdater.java (hunk 2 above)
Comment (aliassheikh) [reply src/main/java/nl/knaw/dans/managedeposit/core/service/DepositStatusUpdater.java]: Will check whether it is going te run in a multithread environment.

**Pull request merged**